src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -0,0 +1,230 @@
+import com.powsybl.sensitivity.SensitivityValue;
+import com.powsybl.sensitivity.factors.BranchFlowPerInjectionIncrease;
+import com.powsybl.sensitivity.factors.BranchFlowPerPSTAngle;
+
+import java.util.*;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class DcSensitivityAnalysis extends AbstractSensitivityAnalysis {
+
+    static class SensitivityVariableConfiguration {
+
+        private final Set<String> busIds;
+
+        private final Set<String> phaseTapChangerHolderIds;
+
+        SensitivityVariableConfiguration(Set<String> busIds, Set<String> phaseTapChangerHolderIds) {
+            this.busIds = Objects.requireNonNull(busIds);
+            this.phaseTapChangerHolderIds = Objects.requireNonNull(phaseTapChangerHolderIds);
+            if (busIds.isEmpty() && phaseTapChangerHolderIds.isEmpty()) {
+                throw new IllegalArgumentException("Bus ID and phase tap changer holder ID list is empty");
+            }
+        }
+
+        Set<String> getBusIds() {
+            return busIds;
+        }
+
+        Set<String> getPhaseTapChangerHolderIds() {
+            return phaseTapChangerHolderIds;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            SensitivityVariableConfiguration that = (SensitivityVariableConfiguration) o;
+            return busIds.equals(that.busIds) && phaseTapChangerHolderIds.equals(that.phaseTapChangerHolderIds);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(busIds);
+        }
+    }
+
+    static class SensitivityFactorGroup {
+
+        private final List<SensitivityFactor<?, ?>> factors = new ArrayList<>();
+
+        private int index = -1;
+
+        List<SensitivityFactor<?, ?>> getFactors() {
+            return factors;
+        }
+
+        int getIndex() {
+            return index;
+        }
+
+        void setIndex(int index) {
+            this.index = index;
+        }
+    }
+
+    public DcSensitivityAnalysis(MatrixFactory matrixFactory) {
+        super(matrixFactory);
+    }
+
+    private SensitivityValue createBranchSensitivityValue(LfNetwork lfNetwork, EquationSystem equationSystem, String branchId,
+                                                          SensitivityFactor<?, ?> factor, SensitivityFactorGroup factorGroup,
+                                                          DenseMatrix states) {
+        LfBranch lfBranch = lfNetwork.getBranchById(branchId);
+        ClosedBranchSide1DcFlowEquationTerm p1 = equationSystem.getEquationTerm(SubjectType.BRANCH, lfBranch.getNum(), ClosedBranchSide1DcFlowEquationTerm.class);
+        double value = Math.abs(p1.calculate(states, factorGroup.getIndex()) * PerUnit.SB);
+        return new SensitivityValue(factor, value, 0, 0);
```

Comment:
Do we not care about function reference and variable reference, or is this a todo ?

Reply:
Yes to do